kv/txn_scope_var.go

```diff
@@ -27,12 +27,11 @@ type TxnScopeVar struct {
 }
 
 // GetTxnScopeVar gets TxnScopeVar from config
-func GetTxnScopeVar() TxnScopeVar {
-	isGlobal, location := config.GetTxnScopeFromConfig()
-	if isGlobal {
-		return NewGlobalTxnScopeVar()
+func GetTxnScopeVar(enableLocalTxn bool) TxnScopeVar {
+	if location := config.GetTxnScopeFromConfig(); enableLocalTxn && location != GlobalTxnScope {
+		return NewLocalTxnScopeVar(location)
 	}
-	return NewLocalTxnScopeVar(location)
+	return NewGlobalTxnScopeVar()
```

Comment:
I don't think introducing `enableLocalTxn` into `GetTxnScopeVar` is a good way. This makes the value of the txn_scope more complex. I think we should to control the switch during `NewTxn` ( for begin transaction) and `PrepareTxnCtx` (for auto-commit).

Reply:
You are right. I decouple these two system variables and make `EnableLocalTxn` only affects `CheckAndGetTxnScope`. PTAL.